src/query/dblink_scan.c

```diff
@@ -120,39 +112,101 @@ dblink_get_basic_utype (T_CCI_U_EXT_TYPE u_ext_type)
+      error = db_make_float (cci_value, *(float *) val);
+      break;
+    case CCI_U_TYPE_DOUBLE:
+      error = db_make_double (cci_value, *(double *) val);
+      break;
+    case CCI_U_TYPE_MONETARY:
+      error = db_make_monetary (cci_value, db_get_currency_default (), *(double *) val);
+      break;
+    case CCI_U_TYPE_STRING:
+      error = db_make_varchar (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    case CCI_U_TYPE_VARNCHAR:
+      error = db_make_varnchar (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    case CCI_U_TYPE_CHAR:
+      error = db_make_char (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    case CCI_U_TYPE_NCHAR:
+      error = db_make_nchar (cci_value, prec, (DB_CONST_C_CHAR) val, len, LANG_SYS_CODESET, LANG_SYS_COLLATION);
+      break;
+    default:
+      assert (false);
+      break;
+    }
+
+  return error;
+}
+
+static int
 dblink_make_date_time (T_CCI_U_TYPE utype, DB_VALUE * value_p, T_CCI_DATE * date_time)
 {
   DB_TIME t_time;
   DB_DATE t_date;
   DB_DATETIME t_datetime;
   DB_TIMESTAMP t_timestamp;
+  int error;
 
   switch (utype)
     {
     case CCI_U_TYPE_TIME:
-      db_make_time (value_p, date_time->hh, date_time->mm, date_time->ss);
+      error = db_make_time (value_p, date_time->hh, date_time->mm, date_time->ss);
       break;
     case CCI_U_TYPE_DATE:
-      db_make_date (value_p, date_time->mon, date_time->day, date_time->yr);
+      error = db_make_date (value_p, date_time->mon, date_time->day, date_time->yr);
       break;
     case CCI_U_TYPE_DATETIME:
-      db_datetime_encode (&t_datetime, date_time->mon, date_time->day, date_time->yr,
-			  date_time->hh, date_time->mm, date_time->ss, date_time->ms);
-      db_make_datetime (value_p, &t_datetime);
+      error = db_datetime_encode (&t_datetime, date_time->mon, date_time->day, date_time->yr,
+				  date_time->hh, date_time->mm, date_time->ss, date_time->ms);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_make_datetime (value_p, &t_datetime);
       break;
     case CCI_U_TYPE_TIMESTAMP:
-      db_time_encode (&t_time, date_time->hh, date_time->mm, date_time->ss);
-      db_date_encode (&t_date, date_time->mon, date_time->day, date_time->yr);
-      db_timestamp_encode (&t_timestamp, &t_date, &t_time);
-      db_make_timestamp (value_p, t_timestamp);
+      error = db_time_encode (&t_time, date_time->hh, date_time->mm, date_time->ss);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_date_encode (&t_date, date_time->mon, date_time->day, date_time->yr);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_timestamp_encode (&t_timestamp, &t_date, &t_time);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_make_timestamp (value_p, t_timestamp);
       break;
     default:
+      assert (false);
```

Comment:
```suggestion
      error = ER_FAILED; // proper error, I just set dummy one for code review.
      assert (false);
```

Reply:
it must not be reached because this routine is called with only the proper data/time type from dblink_scan_next.
it should return an error, we need a proper error number.